### PR TITLE
Add With() to logger

### PR DESCRIPTION
**Pull request by tock-ibm**
Signed-off-by: Yoav Tock <tock@il.ibm.com>

### CODE REVIEW

File: pkg/logger/logger_test.go [hunk 1]
```diff
@@ -298,3 +298,114 @@ func TestErrorPath(t *testing.T) {
...
+
+	tests := []struct {
+		level               string
+		fileName            string
+		expectedMessages    []string
+		notExpectedMessages []string
+	}{
+		{
+			level:    "debug",
+			fileName: "/tmp/debug.txt",
```
Comment (gennadylaventman) [inline pkg/logger/logger_test.go:320]: Can we use here some temp location and not go direct to /tmp folder

File: pkg/logger/logger_test.go [hunk 2]
```diff
@@ -298,3 +298,114 @@ func TestErrorPath(t *testing.T) {
...
+			expectedMessages: []string{
+				"debug message is logged\t{\"id\": \"node1\"}",
```
Comment (cendhu) [inline pkg/logger/logger_test.go:322]: is it possible to prefix the `{"id": "node1"}` with the log message?

Comment (tock-ibm) [reply pkg/logger/logger_test.go:333]: not really... or no that I know of, it's a feature of zap

File: pkg/logger/logger_test.go (hunk 1 above)
Comment (tock-ibm) [reply pkg/logger/logger_test.go]: Done

**Pull request merged**